File: performance_scorecard/performance_scorecard/report/performance_analytics/performance_analytics.py
import frappe

def execute(filters=None):
	columns = [
		{"fieldname": "employee", "label": "Employee", "fieldtype": "Link", "options": "Employee", "width": 150},
		{"fieldname": "department", "label": "Department", "fieldtype": "Link", "options": "Department", "width": 150},
		{"fieldname": "overall_score", "label": "Overall Score", "fieldtype": "Float", "width": 100},
		{"fieldname": "status", "label": "Status", "fieldtype": "Data", "width": 100}
	]
	
	data = []
	
	conditions = ""
	if filters.get("period_start"):
		conditions += f" AND start_date >= '{filters.get('period_start')}'"
	if filters.get("period_end"):
		conditions += f" AND end_date <= '{filters.get('period_end')}'"
		
	scorecards = frappe.db.sql(f"""
		SELECT employee, department, overall_score, status
		FROM `tabPerformance Scorecard`
		WHERE docstatus < 2 {conditions}
	""", as_dict=1)
	
	data = scorecards
	
	return columns, data

File: performance_scorecard/performance_scorecard/report/performance_analytics/performance_analytics.js
frappe.query_reports["Performance Analytics"] = {
    "filters": [
        {
            "fieldname": "company",
            "label": __("Company"),
            "fieldtype": "Link",
            "options": "Company",
            "default": frappe.defaults.get_user_default("Company"),
            "reqd": 1
        },
        {
            "fieldname": "period_start",
            "label": __("Period Start"),
            "fieldtype": "Date",
            "reqd": 1
        },
        {
            "fieldname": "period_end",
            "label": __("Period End"),
            "fieldtype": "Date",
            "reqd": 1
        }
    ]
};
